spark/src/main/java/org/datacleaner/spark/SparkJobLauncher.java

```diff
@@ -0,0 +1,115 @@
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.spark;
+
+import java.io.InputStream;
+
+import org.apache.metamodel.DataContext;
+import org.apache.metamodel.csv.CsvDataContext;
+import org.apache.metamodel.util.HdfsResource;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.datacleaner.configuration.DataCleanerConfiguration;
+import org.datacleaner.configuration.JaxbConfigurationReader;
+import org.datacleaner.connection.Datastore;
+import org.datacleaner.job.AnalysisJob;
+import org.datacleaner.job.JaxbJobReader;
+
+public class SparkJobLauncher {
+
+    private final DataCleanerConfiguration _dataCleanerConfiguration;
+
+    public SparkJobLauncher(DataCleanerConfiguration dataCleanerConfiguration) {
+        _dataCleanerConfiguration = dataCleanerConfiguration;
+    }
+
+    public SparkJobLauncher(String confXmlPath) {
+        _dataCleanerConfiguration = readDataCleanerConfiguration(new HdfsResource(confXmlPath));
+    }
+
+    public void launchJob(AnalysisJob analysisJob) {
+        final String datastoreName = analysisJob.getDatastore().getName();
+
+        final Datastore datastore = _dataCleanerConfiguration.getDatastoreCatalog().getDatastore(datastoreName);
+        if (datastore == null) {
+            throw new IllegalStateException("Datastore referred by the job (" + datastoreName
+                    + ") has not been found in the specified DataCleanerConfiguration");
+        } else {
+            JavaSparkContext sc = null;
+            try {
+                DataContext dataContext = datastore.openConnection().getDataContext(); 
+                if (dataContext instanceof CsvDataContext) {
+                    CsvDataContext csvDataContext = (CsvDataContext) dataContext;
+                    String datastoreFilePath = csvDataContext.getResource().getQualifiedPath();
+            
+                    SparkConf conf = new SparkConf().setAppName("DataCleaner-spark").setMaster("yarn-client");
+                    sc = new JavaSparkContext(conf);
+
+                    JavaRDD<String> instructionsRDD = sc.textFile(datastoreFilePath);
+                    long lineCount = instructionsRDD.count();
+        
+                    System.out.println("Line count: " + lineCount);
```

Comment:
Yeah, the Spark job was not supposed to do anything for now, so I just counted the lines in the input file and printed it. We can leave it as it is or just delete above two lines and really make it do nothing.